### PR TITLE
Fix buying the correct license selected

### PR DESCRIPTION
**Pull request by ClementPasteau**
Moved the Purchase dialogs inside the Information pages as this is where there're only used.
Fix using the correct selected UsageType

### CODE REVIEW

File: newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js [hunk 1]
```diff
@@ -130,45 +142,36 @@ const PrivateAssetPackPurchaseDialog = ({
     []
   );
 
-  const checkUserPurchases = React.useCallback(
```
Comment (ClementPasteau) [inline newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js:133]: simplify logic like on the gametemplate purchase dialog, the purchases are now stored in the context, so we just have to read them

File: newIDE/app/src/AssetStore/PrivateGameTemplates/PrivateGameTemplateInformationDialog.js [hunk 2]
```diff
@@ -34,7 +30,6 @@ const PrivateGameTemplateInformationDialog = ({
           label={<Trans>Back</Trans>}
           primary={false}
           onClick={onClose}
-          disabled={isPurchaseDialogOpen}
```
Comment (ClementPasteau) [inline newIDE/app/src/AssetStore/PrivateGameTemplates/PrivateGameTemplateInformationDialog.js:37]: this wasn't really useful as a dialog is on top of this one anyway

File: newIDE/app/src/AssetStore/index.js [hunk 3]
```diff
@@ -452,44 +442,6 @@ export const AssetStore = React.forwardRef<Props, AssetStoreInterface>(
       [shopNavigationState, saveScrollPosition]
     );
 
-    // If the user has received the pack they are currently viewing,
-    // we update the window to show it if they are not already on the pack page.
-    React.useEffect(
```
Comment (ClementPasteau) [inline newIDE/app/src/AssetStore/index.js:457]: This is not needed anymore as the Information Page now shows correctly that the pack is "Owned" with a button to browse the assets. (better ux than redirecting to the assets list and having to wait for them to load)

File: newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js [hunk 4]
```diff
@@ -130,45 +142,36 @@ const PrivateAssetPackPurchaseDialog = ({
...
+          isPurchasing &&
+          assetPackPurchases &&
+          assetPackPurchases.find(
             userPurchase =>
               userPurchase.productId === privateAssetPackListingData.id
           )
         ) {
-          // We found the purchase, the user has bought the asset pack.
-          // We do not close the dialog yet, as we need to trigger a refresh of the asset store.
+          // We found the purchase, the user has bought the game template.
```
Comment (AlexandreSi) [inline newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js:156]:
```suggestion
          // We found the purchase, the user has bought the asset pack.
```

File: newIDE/app/src/AssetStore/index.js (hunk 3 above)
Comment (4ian) [reply newIDE/app/src/AssetStore/index.js:457]: Good riddance

**Pull request merged**